firedrake/slate/slac/compiler.py

```diff
@@ -246,12 +244,12 @@ def compile_expression(slate_expr, tsfc_parameters=None):
         args.append(ast.Decl("int **", cell_orientations))
 
     # Coefficient information
-    for c in expr_coeffs:
+    for c, csym_info in iteritems(builder.coefficient_map):
         if isinstance(c, Constant):
             ctype = "%s *" % SCALAR_TYPE
         else:
             ctype = "%s **" % SCALAR_TYPE
-        args.extend([ast.Decl(ctype, csym) for csym in builder.coefficient(c)])
+        args.extend([ast.Decl(ctype, csym) for csym in csym_info])
```

Comment:
So previously you pulled the coefficients out of the expression and then called `builder.coefficient` here.  Now you make the coefficient_map in the constructor of the kernel builder and then iterate over that.  But now `coefficient_map` is a public API member.  Why not iterate over the coefficients of the expression and call `builder.coefficient` as before?